src/transformers/models/udop/processing_udop.py

```diff
@@ -57,28 +90,10 @@ def __call__(
         self,
         images: Optional[ImageInput] = None,
         text: Union[TextInput, PreTokenizedInput, List[TextInput], List[PreTokenizedInput]] = None,
-        text_pair: Optional[Union[PreTokenizedInput, List[PreTokenizedInput]]] = None,
```

Comment:
Agree they don't change tokenization, however I think they belong there though, just because they were present before and are part of the `tokenizer` signature: from udop docs we say 

>  Additionally, it also supports passing `text_target` and `text_pair_target` to the tokenizer, which can be used to
>     prepare labels for language modeling tasks.  
so even if it does not explicitly change the text, it's still a tokenizer option so in terms of separation of concerns, for me it belongs here! (mostly, because it was there before)

Reply:
Yes I can see how that would be a problem for backward compatibility. Maybe we should deprecate the use of text_pair, text_target etc. as args and not kwargs? Especially since they are optional and other kwargs can be used without them (e.g. `inputs = processor(image, words, boxes=boxes, return_tensors="pt")` in `UdopModel` doc). However I'm not sure how we could catch the use of too many args to provide a deprecation warning.

Reply:
I am not very strong about it, we can keep it as TextKwargs as long as we don't break BC

Reply:
I've done this which is a bit hacky but should preserve BC:
https://github.com/huggingface/transformers/blob/6d7e086acdb9bd4f8fd4ea03a3615c6b8e212519/src/transformers/models/udop/processing_udop.py#L117-L123

Reply:
Cool, let's do it logger,warning_once and move the warning below, so that users see it only if passing text_pair without indicating `text_pair=my_text`

Reply:
Honestly not a big fan of this - we shouldn't be using kwargs for a hack for which they are not advertised. Will take a look this afternoon and try to suggest something

Reply:
well I'm kind of out of ideas, I'll trust you on finding something clean, at worst we can do as you say but add a deprecation cycle for a few versions later. Closest I could find that does modify the signature is simply by capturing all extra args, like so

```python
    def __call__(
        self,
        images: Optional[ImageInput] = None,
        text: Union[TextInput, PreTokenizedInput, List[TextInput], List[PreTokenizedInput]] = None,
        *args,
        audio=None,
        videos=None,
        **kwargs: Unpack[UdopProcessorKwargs],
    ) -> BatchFeature:
        """
        This method first forwards the `images` argument to [`~UdopImageProcessor.__call__`]. In case
        [`UdopImageProcessor`] was initialized with `apply_ocr` set to `True`, it passes the obtained words and
        bounding boxes along with the additional arguments to [`~UdopTokenizer.__call__`] and returns the output,
        together with the prepared `pixel_values`. In case [`UdopImageProcessor`] was initialized with `apply_ocr` set
        to `False`, it passes the words (`text`/``text_pair`) and `boxes` specified by the user along with the
        additional arguments to [`~UdopTokenizer.__call__`] and returns the output, together with the prepared
        `pixel_values`.

        Alternatively, one can pass `text_target` and `text_pair_target` to prepare the targets of UDOP.

        Please refer to the docstring of the above two methods for more information.
        """
        # verify input
        output_kwargs = self._merge_kwargs(
            UdopProcessorKwargs,
            tokenizer_init_kwargs=self.tokenizer.init_kwargs,
            **kwargs,
        )
        # for BC, handle unexpected positional arguments
        if len(args) > 0:
            logger.warning_once(
                f"Received unexpected positional arguments. These will be mapped accordingly to `text_pair`."
            )
            if len(args) == 1:
                # if there's one extra positional argument, assume it's `text_pair` for backward compatibility
                output_kwargs['text_kwargs']['text_pair'] = args[0]
```
which feels a bit less hacky if it indeed works. It would also allow to not add extra placeholder args when we have more than one extra arg to take care of, as in this cool work: https://github.com/huggingface/transformers/pull/32180
However I will take my annual holidays soon so I won't be able to decide more on that, I'll trust you to move on with something incredible anyways as you've done amazing work already @yonigozlan @zucchini-nlp :purple_heart: